### ISSUE TITLE
Bug with signOutWithCompletion

### ISSUE THREAD

**Issue by gerchicov-bp** [OneDrive/onedrive-sdk-ios#107]
`signOutWithCompletion` twice causes app to crash

**Comment by kevklam** [comment]
Please refer to the following pattern:

``` objc
[self.client signOutWithCompletion:^(NSError *signOutError) {
    self.client = nil;
    // perform any other post-signout actions
}
```

Once a client has been signed out, it is considered disposed and should not be used again. Instead use **[ODClient clientWithCompletion:]** or **[ODClient authenticatedClientWithCompletion:]** to create a new client.

**Comment by gerchicov-bp** [comment]
The problem is completion block is executed after the second call of `signOutWithCompletion:` so your code won't work. Only this code works:
`[self.client signOutWithCompletion:^(NSError *signOutError) {
    // perform any other post-signout actions
}
self.client = nil;`

But it seems it doesn't conform to your library style

**Comment by kevklam** [comment]
Thanks, will investigate.

**Comment by kevklam** [comment]
Should be fixed with #116
